security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -582,9 +582,25 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 
 		// Re-generate secret if it's expired.
 		if sc.shouldRotate(&secret) {
+			// withToken indicates whether we use the token in our CSR request
+			withToken := true
```

Comment:
you can just `withToken := sc.secOpts.ProvCert == ""` and then in line 593 you can just `if !withToken { ... }`, also move this variable to be just above the if where it is used.

Reply:
withToken will never be used. in the next revision